### PR TITLE
65767 - Replace appointment start field with localStartTime field.#65767Replace appointment start field with localStartTime field.

### PR DESCRIPTION
**Pull request by vbahinwillit**
## Summary
This PR replaces the appointment 'start' field with the 'localStartTime' field.

## Related issue(s)

department-of-veterans-affairs/va.gov-team#65767

## Testing done
Unit and e2e testing.

## Screenshots
N/A

## What areas of the site does it impact?
VAOS

## Acceptance criteria
All test should pass

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/vaos/tests/appointment-list/components/ConfirmedAppointmentDetailsPage/index.unit.spec.jsx [hunk 1]
```diff
@@ -356,10 +360,7 @@ describe('VAOS <ConfirmedAppointmentDetailsPage> with VAOS service', () => {
     expect(
       screen.getByRole('heading', {
         level: 1,
-        name: new RegExp(
-          futureDate.tz('America/Denver').format('dddd, MMMM D, YYYY'),
-          'i',
-        ),
+        name: new RegExp(futureDate.format('dddd, MMMM D, YYYY'), 'i'),
```
[the remark this answers is not in the document]
Comment (vbahinwillit) [reply src/applications/vaos/tests/appointment-list/components/ConfirmedAppointmentDetailsPage/index.unit.spec.jsx:363]: I don't think so. My assumption is that the start date has already been converted to the timezone of the facility

**Comment by vbahinwillit** [comment]
> @vbahinwillit - I found some components that are still using the appointment.start as the source of truth for displaying appointment start time.
> 
> * `ConfirmationPageV2` in `vets-website/src/applications/vaos/covid-19-vaccine/components/ConfirmationPageV2.jsx`.

@simiadebowale - I don't see where appointment.start is being used in ConfirmationPageV2.

**Comment by simiadebowale** [comment]
> > @vbahinwillit - I found some components that are still using the appointment.start as the source of truth for displaying appointment start time.
> > 
> > * `ConfirmationPageV2` in `vets-website/src/applications/vaos/covid-19-vaccine/components/ConfirmationPageV2.jsx`.
> 
> @simiadebowale - I don't see where appointment.start is being used in ConfirmationPageV2.

@vbahinwillit - I noticed that file is still converting the timezone to the facility Timezone using `getTimezoneByFacilityId`

**Comment by vbahinwillit** [comment]
> > > @vbahinwillit - I found some components that are still using the appointment.start as the source of truth for displaying appointment start time.
> > > 
> > > * `ConfirmationPageV2` in `vets-website/src/applications/vaos/covid-19-vaccine/components/ConfirmationPageV2.jsx`.
> > 
> > 
> > @simiadebowale - I don't see where appointment.start is being used in ConfirmationPageV2.
> 
> @vbahinwillit - I noticed that file is still converting the timezone to the facility Timezone using `getTimezoneByFacilityId`

@simiadebowale - It's converting to the facility timezone because we don't use the return value from the POST call that created the appointment. I am assuming the POST call returns the newly created appointment with the new 'localStartTime' attribute. It seems like another ticket should be created if the goal is to remove all use of 'getTimezoneByFacilityId' since this ticket is for the appointment list (that's the only place the transformer is used).

**Review by simiadebowale — APPROVED** [review]
LGTM!

**Comment by simiadebowale** [comment]
> > > > @vbahinwillit - I found some components that are still using the appointment.start as the source of truth for displaying appointment start time.
> > > > 
> > > > * `ConfirmationPageV2` in `vets-website/src/applications/vaos/covid-19-vaccine/components/ConfirmationPageV2.jsx`.
> > > 
> > > 
> > > @simiadebowale - I don't see where appointment.start is being used in ConfirmationPageV2.
> > 
> > 
> > @vbahinwillit - I noticed that file is still converting the timezone to the facility Timezone using `getTimezoneByFacilityId`
> 
> @simiadebowale - It's converting to the facility timezone because we don't use the return value from the POST call that created the appointment. I am assuming the POST call returns the newly created appointment with the new 'localStartTime' attribute. It seems like another ticket should be created if the goal is to remove all use of 'getTimezoneByFacilityId' since this ticket is for the appointment list (that's the only place the transformer is used).

@vbahinwillit - I will do some digging to determine if we can remove every use of 'getTimezoneByFacilityId'. We likely still need it in the scheduling flow.

**Pull request merged**